hazelcast/src/main/java/com/hazelcast/cache/impl/nearcache/NearCacheManager.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2008-2013, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cache.impl.nearcache;
+
+import com.hazelcast.config.NearCacheConfig;
+
+import java.util.Collection;
+
+/**
+ * {@link NearCacheManager} is the contract point to manage all existing
+ * {@link com.hazelcast.cache.impl.nearcache.NearCache} instances.
+ */
+public interface NearCacheManager {
+
+    /**
+     * Gets the {@link com.hazelcast.cache.impl.nearcache.NearCache} instance
+     * associated with given <code>name</code>.
+     *
+     * @param name the name of the {@link com.hazelcast.cache.impl.nearcache.NearCache} instance will be got
+     * @param <K> the type of the key for Near Cache
+     * @param <V> the type of the value for Near Cache
+     * @return the {@link com.hazelcast.cache.impl.nearcache.NearCache} instance
+     *         associated with given <code>name</code>
+     */
+    <K, V> NearCache<K, V> getNearCache(String name);
+
+    /**
+     * Creates a new {@link com.hazelcast.cache.impl.nearcache.NearCache} with given configurations
+     * or returns existing one.
+     *
+     * @param name              the name of the {@link com.hazelcast.cache.impl.nearcache.NearCache}
+     *                          to be created or existing one
+     * @param nearCacheConfig   the {@link com.hazelcast.config.NearCacheConfig} of the
+     *                          {@link com.hazelcast.cache.impl.nearcache.NearCache} to be created
+     * @param nearCacheContext  the {@link com.hazelcast.cache.impl.nearcache.NearCacheContext} of the
+     *                          {@link com.hazelcast.cache.impl.nearcache.NearCache} to be created
+     * @param <K> the key type of the {@link com.hazelcast.cache.impl.nearcache.NearCache}
+     * @param <V> the value type of the {@link com.hazelcast.cache.impl.nearcache.NearCache}
+     * @return the created or existing {@link com.hazelcast.cache.impl.nearcache.NearCache} instance
+     *         associated with given <code>name</code>
+     */
+    <K, V> NearCache<K, V> createNearCacheIfAbsent(String name, NearCacheConfig nearCacheConfig,
```

Comment:
I think `getOrCreateNearCache()` is more convenient name here, we call similar methods as `getOrCreateX()`.